### PR TITLE
Use correct package resource for chef-client version

### PR DESCRIPTION
**Pull request by meldixo1**

### CODE REVIEW

File: resources/r_package.rb [hunk 1]
```diff
@@ -26,10 +26,29 @@ def local_r_version
...
-  package pkg_name do
-    source pkg_source
-    options '/s'
-    action :install
+  # for chef-client 13 and later, use the Chef package resource
+  # for chef-client earlier than 13, use the Chef windows_package resource
+  # chef_version = `chef-client --version`
+
+  # find the chef-client version
+  get_chef_version = Mixlib::ShellOut.new('chef-client --version')
```
Comment (itsamenathan) [inline resources/r_package.rb]: Could you do something like ```node['chef_packages']['chef']['version']``` instead?

**Comment by jakemig** [comment]
Is the `chef_packages` attribute available on the first run? That'd be my only concern.

File: spec/unit/recipes/default_spec.rb [hunk 2]
```diff
@@ -81,3 +82,27 @@
     end
   end
 end
+
+describe 'alteryx-server::default' do
+  context 'When chef-clint is 13' do
```
Comment (jakemig) [inline spec/unit/recipes/default_spec.rb]: clint->client

File: resources/r_package.rb [hunk 3]
```diff
@@ -26,10 +26,26 @@ def local_r_version
   pkg_name = "Alteryx Predictive Tools with R #{version}"
   pkg_source = source
 
-  package pkg_name do
-    source pkg_source
-    options '/s'
-    action :install
+  # for chef-client 13 and later, use the Chef package resource
+  # for chef-client earlier than 13, use the Chef windows_package resource
+  # chef_version = `chef-client --version`
```
Comment (jakemig) [inline resources/r_package.rb]: We can remove this line.

**Pull request merged**